src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+                    return driver.findElement(By.id("createTaskModal"));
+                  }
+                })
+            .isDisplayed();
+
+    // After clicking on the add task button, the modal should be displayed
+    assertTrue("Create task modal should be displayed", createTaskModalElementDisplayed);
+
+    // Inputs task details using Javascript Executor
+    js.executeScript("document.getElementById('task-detail-input').value='" + details + "';");
+    js.executeScript("document.getElementById('task-overview-input').value='" + overview + "';");
+    js.executeScript("document.getElementById('rewarding-point-input').value='" + points + "';");
+    js.executeScript(
+        "document.getElementById('category-input').value='"
+            + TASK_CATEGORIES[categoryIndex]
+            + "';");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    // clicks on submit create task button
+
+    wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(By.id("submit-create-task"));
+              }
+            })
+        .click();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Verifies that newly added task's details are correctly display in userpage's need help table
+   */
+  private void verifyNewTaskUserPage() {
+    // Location of most recent task in user page's need help
+    String taskRowXPath = "//table[@id='need-help']/tbody/tr[1]";
+
+    String rowTaskOverview =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskRowXPath + "/td[1]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("overview"), rowTaskOverview);
+
+    String rowHelper =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskRowXPath + "/td[2]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("helper"), rowHelper);
+
+    String rowStatus =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskRowXPath + "/td[3]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("status"), rowStatus);
+
+    verifyTaskDetails(taskRowXPath + "/td[1]");
+  }
+
+  /** Verifies that newly added tasks are displayed properply in homepage */
+  private void verifyNewTaskHomepage() {
+
+    // First task location in homepage
+    String taskXPath = "//div[@id='tasks-list']/div[1]/div[2]";
+
+    String taskOverviewActual =
```

Comment:
Still we can try to use a for loop to replace code from line 690 to 718. For example, `for (int i = 1; i < 4; i++) ` and when i equals to 2, we can treat it differently.

Reply:
I'm going to keep this one and the other one mentioned in the comment below as is, because of how they don't all behave the same, and making those exceptions in the for loop might make the code not readable at all.

Reply:
Cool.